app/modules/announcements/announcements.py

```diff
@@ -0,0 +1,191 @@
+import logging
+
+import discord
+from discord import Guild as DiscordGuild
+from discord import Member as DiscordMember
+from discord import TextChannel as DiscordTextChannel
+from discord.ext import commands, bridge
+from discord.ext.bridge import BridgeContext
+
+from app import client, config
+from app.cache import cache_get_dict, cache_set_dict
+from app.database.database import get_member, get_guild
+from app.database.models import Member as DatabaseMember
+from app.utilities.cogs import defer
+
+
+class Announcements(commands.Cog):
+    def __init__(self, bot: client.PhoneWave):
+        self.bot = bot
+
+    def get_announcement_channel(self, announcement_channel_id, guild: DiscordGuild):
+        if announcement_channel_id:
+            return self.bot.get_channel(announcement_channel_id)
+        elif guild.system_channel:
+            return guild.system_channel
+        else:
+            return None
+
+    def get_announcement_data(self, guild: DiscordGuild):
+        data = cache_get_dict("announcements", guild=guild)
+
+        if not data:
+            db_guild = get_guild(gid=guild.id)
+            data = {
+                "channel_id": db_guild.announcements.channel_id,
+                "welcome": db_guild.announcements.welcome,
+                "farewell": db_guild.announcements.farewell,
+            }
+            cache_set_dict("announcements", data, guild=guild)
+
+        channel = self.get_announcement_channel(data.get("channel_id"), guild)
+
+        return {
+            "channel": channel,
+            "welcome": data.get("welcome"),
+            "farewell": data.get("farewell"),
+        }
+    
+    def replace_message_template(self, template: str, default: str, member: DiscordMember, db_member: DatabaseMember):
+        if not template:
+            template = default
+
+        message = template.replace("{number}", str(db_member.lab_member_number))  # TODO: add zero-padding, e.g. 001 
+        message = message.replace("{name}", member.mention)
+        return message
+    
+    def create_member(self, member: DiscordMember):
+        db_member = get_member(gid=member.guild.id, uid=member.id)
```

Comment:
You don't need to pass in named arguments for get_member and get_guild, but I'm seeing this everywhere. Is this some Pythonic thing I'm missing? I can understand for lines like:

  `embed.add_field(name="Lab Member", value=f"#{db_member.lab_member_number}", inline=False)`

As it helps with readability and parameter naming, but `get_member(gid=message.guild.id, uid=message.author.id)` is explicit even without the keywords

Reply:
I don't generally remember the order of the arguments, so I just let the IDE suggest them. I'll do a pass to clean it up